Updating Service Bus Templates

wsilveiranz opened the pull request:
- Adding sequential convoy template
- Adding category to all service bus template (design patterns)

wsilveiranz commented:
@divyaswarnkar, @Elaina-Lee please review content for technical accuracy.

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+{
+    "title": "Service Bus Peek Lock Pattern - Renew lock token",
+    "description": "Get new message from a Service Bus queue and lock it for processing, renewing the lock token until processing is completed. The message will be deleted from the queue after the processing is completed. This pattern is useful when you want to ensure that the message is processed only once and not lost in case of a failure during processing.",
```
Consistency edits to mirror descriptions in other templates:

```suggestion
    "description": "This workflow template triggers when a message is available in a Service Bus queue. The workflow gets the message from the queue, locks the message for processing, and renews the lock token until processing completes. The message is deleted from the queue after processing completes. This pattern helps you make sure that the workflow processes the message only once and that the message doesn't get lost if a failure happens during processing.",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+{
+    "title": "Service Bus Peek Lock Pattern - Renew lock token",
```
Consistency with other templates:

```suggestion
    "title": "Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+{
+    "title": "Service Bus Peek Lock Pattern - Renew lock token",
+    "description": "Get new message from a Service Bus queue and lock it for processing, renewing the lock token until processing is completed. The message will be deleted from the queue after the processing is completed. This pattern is useful when you want to ensure that the message is processed only once and not lost in case of a failure during processing.",
+    "prerequisites": "Before you can use this template, you need to configure a Service Bus namespace and create a queue. For more information about Service Bus see [Create a new queue on Service Bus](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal)",
```
- Consistency edits with other templates.
-  We shouldn't hardcode URLs, nor include the locale (en-us) in the link, so we need to replace with an FW link.

```suggestion
    "prerequisites": "Before you can use this template, you need to create or set up a Service Bus namespace with a queue. For more information, see [Create a Service Bus namespace and a queue](https://go.microsoft.com/fwlink/?linkid=2296007).",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+    "parameters": [
+        {
+            "name": "queueName_#workflowname#",
+            "displayName": "Queue Name",
```
- Use sentence casing for parameter names, like the other templates.

```suggestion
            "displayName": "Queue name",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+            "name": "queueName_#workflowname#",
+            "displayName": "Queue Name",
+            "type": "String",
+            "default": "This is the name of the queue from which the message will be fetched.",
```
```suggestion
            "default": "The name of the queue from where to get the message.",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+            "displayName": "Queue Name",
+            "type": "String",
+            "default": "This is the name of the queue from which the message will be fetched.",
+            "description": "This is a parameter",
```
```suggestion
            "description": "The name of the Service Bus queue from where to get the message.",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+        },
+        {
+            "name": "messageBatchSize_#workflowname#",
+            "displayName": "Message Batch Size",
```
```suggestion
            "displayName": "Message batch size",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+            "displayName": "Message Batch Size",
+            "type": "Int",
+            "default": "50",
+            "description": "This indicates how many messages will be fetched in a single batch.",
```
```suggestion
            "description": "The number of messages to get as a single batch.",

```

ecfan commented on sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,60 @@
+            "displayName": "Delay before next token renewal",
+            "type": "Int",
+            "default": "3",
+            "description": "This indicates how long to wait for processing before the lock token is renewed. This value should be smaller than the lock token duration of the queue.",
```
```suggestion
            "description": "The number of minutes to delay processing before renewing the lock token. Make sure this value is smaller than the queue's lock token duration.",

```

ecfan reviewed (changes requested):
@wsilveiranz: Feedback to make UX content similar to other templates for consistency.

ecfan commented:
#assign:ecfan

wsilveiranz commented:
Thanks @ecfan - I've accepted all the suggestions.

ecfan commented:
@wsilveiranz: Can we please change the folder to look like the others: 

get-message-service-bus-queue-peek-lock-renew-sequential-convoy

wsilveiranz commented:
> @wsilveiranz: Can we please change the folder to look like the others:
> 
> get-message-service-bus-queue-peek-lock-renew-sequential-convoy

Done!

ecfan commented:
[celebrate]     Esther Fan reacted to your message:
________________________________
From: Wagner Silveira ***@***.***>
Sent: Thursday, October 31, 2024 7:29:57 AM
To: Azure/LogicAppsTemplates ***@***.***>
Cc: Esther Fan ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/LogicAppsTemplates] Updating Service Bus Templates (PR #51)


@wsilveiranz<https://github.com/wsilveiranz>: Can we please change the folder to look like the others:

get-message-service-bus-queue-peek-lock-renew-sequential-convoy

Done!

—
Reply to this email directly, view it on GitHub<https://github.com/Azure/LogicAppsTemplates/pull/51#issuecomment-2449204859>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADEYJD7KQ4UU3QI6BO55S53Z6HL7LAVCNFSM6AAAAABQ5KXGU6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINBZGIYDIOBVHE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ecfan commented on get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,69 @@
+{
+    "title": "Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern",
```
```suggestion
    "title": "Azure Service Bus (sequential convoy): Get message from a queue with token renewal - Peek lock renew pattern",

```

ecfan replied:
This version could work too: 

"Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern (sequential convoy)"

ecfan commented:
[celebrate]     Esther Fan reacted to your message:
________________________________
From: Wagner Silveira ***@***.***>
Sent: Thursday, October 31, 2024 5:37:12 AM
To: Azure/LogicAppsTemplates ***@***.***>
Cc: Esther Fan ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/LogicAppsTemplates] Updating Service Bus Templates (PR #51)


Thanks @ecfan<https://github.com/ecfan> - I've accepted all the suggestions.

—
Reply to this email directly, view it on GitHub<https://github.com/Azure/LogicAppsTemplates/pull/51#issuecomment-2449042338>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADEYJD25RKO6LVHI4YPP3B3Z6G6YRAVCNFSM6AAAAABQ5KXGU6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINBZGA2DEMZTHA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ecfan commented on get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:
```diff
@@ -1,9 +1,9 @@
+    "title": "Azure Service Bus: Sequential convoy pattern",
 
-    "description": "This workflow template triggers when a message is available in a Service Bus queue. The workflow gets the message from the queue, locks the message for processing, and renews the lock token until processing completes. The message is deleted from the queue after processing completes. This pattern helps you make sure that the workflow processes the message only once and that the ...
+    "description": "This workflow template triggers when a message is available in a Service Bus session enabled queue. The workflow processes related messages from a specific session on the same workflow instance, creating different workflow instances for each session in the queue.",
```
I'm having trouble parsing some of this. I'm not sure which specific item creates difference workflow instances for each session in the queue?

ecfan commented on get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:
```diff
@@ -1,9 +1,9 @@
+    "description": "This workflow template triggers when a message is available in a Service Bus session enabled queue. The workflow processes related messages from a specific session on the same workflow instance, creating different workflow instances for each session in the queue.",
 
-    "prerequisites": "Before you can use this template, you need to create or set up a Service Bus namespace with a queue. For more information, see [Create a Service Bus namespace and a queue](https://go.microsoft.com/fwlink/?linkid=2296007).",
+    "prerequisites": "Before you can use this template, you need to configure a Service Bus namespace and create a queue with session enabled. For more information about Service Bus see [Create a new queue on Service Bus](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal). You also need to configure your Logic App to support stateful connectors, which is  ...
```
@wsilveiranz: Was there a reason for not using the previously provided FW link? We shouldn't be using hardcoded doc URLs with the locale (en-us) included.

```suggestion
    "prerequisites": "Before you can use this template, you need to set up a Service Bus namespace and create a queue with sessions enabled. For more information, see [Create a Service Bus namespace and a queue](https://go.microsoft.com/fwlink/?linkid=2296007) and [Enable message sessions for a Service Bus queue](https://go.microsoft.com/fwlink/?linkid=2296202). You also need to set up your logic app to support stateful connectors, which is required for sessions. For more information, see [Enable stateful mode for stateless built-in connectors in Azure Logic Apps](https://go.microsoft.com/fwlink/?linkid=2296010).",

```

wsilveiranz replied:
The first link was a copy and paste error.

ecfan commented on get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:
```diff
@@ -1,9 +1,9 @@
 {
-    "title": "Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern",
+    "title": "Azure Service Bus: Sequential convoy pattern",
```
This title appears on the gallery browse page and should be more descriptive for customers who don't know what "sequential convoy" means as well as those who do. Also use word structure similar to other template titles (verb-first wording).

```suggestion
    "title": "Azure Service Bus: Process related messages from a session-enabled queue for the same workflow instance - Sequential convoy pattern",

```

wsilveiranz commented:
@ecfan, I think some of my commits overrode yours, but I updated them now to align. I've also updated the description to try and clarify the process. Let me know if you are ok with that.

wsilveiranz commented on get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:
```diff
@@ -0,0 +1,70 @@
+{
+    "title": "Azure Service Bus: Process related messages from a session-enabled queue for the same workflow instance - Sequential convoy pattern",
+
+    "description": "This workflow template triggers when a message is available in a Service Bus session enabled queue. The workflow creates a new instances for each session available in the queue, processes related messages from that session in the same workflow instance, one at a time.",
```
Exactly. You nailed it!

ecfan reviewed (approved):
Thanks!
